Source/Particles/Gather/GetExternalFields.H

```diff
@@ -49,6 +61,44 @@ struct GetExternalField
             field_y += m_yfield_partparser(x, y, z, m_time);
             field_z += m_zfield_partparser(x, y, z, m_time);
         }
+        else if (m_type == RepeatedPlasmaLens)
```

Comment:
I am surprised this works, isn't `m_type` a host-side member variable?

Reply:
Yes, in this case, the whole object is copied to the GPU.